lib/src/dom/within.dart

```diff
@@ -50,12 +48,12 @@ class WithinQueries extends ScopedQueries {
 /// > See: <https://testing-library.com/docs/dom-testing-library/api-within>
 ///
 /// {@category Queries}
-WithinQueries within(Node node) {
+WithinQueries within(Node? node) {
   if (node == null) {
     throw ArgumentError.notNull('node');
   }
 
-  if (node is! ShadowRoot && !isOrContains(document.body, node)) {
+  if (node is! ShadowRoot && !isOrContains(document.body!, node)) {
```

Comment:
Instead of forcing non-null for `document.body` here, should we instead add a condition?

```suggestion
  if (node is! ShadowRoot && document.body != null && !isOrContains(document.body, node)) {
```

Reply:
I think in a Dart unit test environment, it's pretty safe to assume that document.body isn't null by the time the tests run, so `!` seems fine to me.

Another option is [Node.isConnected](https://developer.mozilla.org/en-US/docs/Web/API/Node/isConnected). We can't use it in prod code due to IE11 (though we could probably polyfill it), but since this is within unit tests we can use it.

Weird thing is it's `bool?` in Dart for some reason, so you have to null-aware it
```suggestion
  if (node is! ShadowRoot && !(node.isConnected ?? false)) {
```